### PR TITLE
opentelemetry-cpp 1.2.0 (new formula)

### PR DESCRIPTION
**Pull request by flier**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/opentelemetry-cpp.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                    "-DBUILD_TESTING=OFF",
+                    "-DWITH_EXAMPLES=OFF",
+                    "-DWITH_JAEGER=ON",
+                    "-DWITH_LOGS_PREVIEW=ON",
+                    "-DWITH_METRICS_PREVIEW=ON",
+                    "-DWITH_OTLP_GRPC=ON",
+                    "-DWITH_OTLP_HTTP=ON",
+                    "-DWITH_OTLP=ON",
+                    "-DWITH_PROMETHEUS=ON"
+    system "make", "install"
```
Comment (carlocab) [inline Formula/opentelemetry-cpp.rb:30]:
```suggestion
    system "cmake", "-S", ".", "-B", "build",
                    "-DCMAKE_INSTALL_RPATH=#{rpath}",
                    "-DBUILD_TESTING=OFF",
                    "-DWITH_EXAMPLES=OFF",
                    "-DWITH_JAEGER=ON",
                    "-DWITH_LOGS_PREVIEW=ON",
                    "-DWITH_METRICS_PREVIEW=ON",
                    "-DWITH_OTLP_GRPC=ON",
                    "-DWITH_OTLP_HTTP=ON",
                    "-DWITH_OTLP=ON",
                    "-DWITH_PROMETHEUS=ON"
                    *std_cmake_args
    system "cmake", "--build", "build"
    system "cmake", "--install", "build"
```

**Review by carlocab — COMMENTED** [review]
Linux might need additional `-l` flags.

**Comment by flier** [comment]
> Linux might need additional `-l` flags.

Changing the order of the linked libraries seems to fix the problem.

**Comment by carlocab** [comment]
Ah yes. That too. GNU `ld` is very picky about the order of the `-l` flags.

**Review by carlocab — COMMENTED** [review]
Please rebase (with `git rebase`) your PR branch to remove the merge commit. When you do so, please squash all your commits together and use the commit message
```
opentelemetry-cpp 1.2.0 (new formula)
```
See the commit style guide linked in the PR template for details.

File: Formula/opentelemetry-cpp.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        // Set the global trace provider
+        trace_api::Provider::SetTracerProvider(provider);
+      }
+    EOS
+    system ENV.cxx, "test.c", "-std=c++11", "-I#{include}", "-L#{lib}",
+           "-lopentelemetry_resources",
+           "-lopentelemetry_trace",
+           "-lopentelemetry_exporter_ostream_span",
+           "-lopentelemetry_common",
+           "-o", "simple-example"
```
Comment (carlocab) [inline Formula/opentelemetry-cpp.rb:64]: Looks like you need a `-pthread` (or `-lpthread`?) somewhere here

Comment (flier) [reply Formula/opentelemetry-cpp.rb]: Okay, it looks like older versions of gcc don't behave the same way.

File: Formula/opentelemetry-cpp.rb [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    # cmake install misses a header open-telemetry/opentelemetry-cpp#1273
+    patch do
+      url "https://github.com/open-telemetry/opentelemetry-cpp/commit/e072daa229f72431b1c50986645d51180a586d09.patch?full_index=1"
+      sha256 "ce54f49532f81e6bc984d8e2e2436d96be1abd2f819f42029168f45aa7b068ed"
+    end
+  end
+
+  head do
+    url "https://github.com/open-telemetry/opentelemetry-cpp.git", branch: "main"
+  end
```
Comment (carlocab) [inline Formula/opentelemetry-cpp.rb:19]:
```suggestion
  license "Apache-2.0"
  head "https://github.com/open-telemetry/opentelemetry-cpp.git", branch: "main"

  stable do
    url "https://github.com/open-telemetry/opentelemetry-cpp/archive/refs/tags/v1.2.0.tar.gz"
    sha256 "7a6420f9e4fa44b81a5b06e30e5e116da71decc9086e5cc4f126e1efc8a397c2"

    # cmake install misses a header open-telemetry/opentelemetry-cpp#1273
    patch do
      url "https://github.com/open-telemetry/opentelemetry-cpp/commit/e072daa229f72431b1c50986645d51180a586d09.patch?full_index=1"
      sha256 "ce54f49532f81e6bc984d8e2e2436d96be1abd2f819f42029168f45aa7b068ed"
    end
  end
```
We don't need a `head` block that consists of a single line.

File: Formula/opentelemetry-cpp.rb [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+  stable do
+    url "https://github.com/open-telemetry/opentelemetry-cpp/archive/refs/tags/v1.2.0.tar.gz"
+    sha256 "7a6420f9e4fa44b81a5b06e30e5e116da71decc9086e5cc4f126e1efc8a397c2"
+
+    # cmake install misses a header open-telemetry/opentelemetry-cpp#1273
+    patch do
+      url "https://github.com/open-telemetry/opentelemetry-cpp/commit/e072daa229f72431b1c50986645d51180a586d09.patch?full_index=1"
+      sha256 "ce54f49532f81e6bc984d8e2e2436d96be1abd2f819f42029168f45aa7b068ed"
+    end
```
Comment (carlocab) [inline Formula/opentelemetry-cpp.rb:14]: We don't accept new formulae that need patches. Let's just wait for the next release that includes this.

**Pull request closed without merging**